core/capabilities/ccip/ccipaptos/commitcodec.go

```diff
@@ -0,0 +1,192 @@
+package ccipaptos
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+
+	"github.com/aptos-labs/aptos-go-sdk"
+	"github.com/aptos-labs/aptos-go-sdk/bcs"
+
+	cciptypes "github.com/smartcontractkit/chainlink-ccip/pkg/types/ccipocr3"
+)
+
+// CommitPluginCodecV1 is a codec for encoding and decoding commit plugin reports.
+// Compatible with ccip::offramp version 1.6.0
+type CommitPluginCodecV1 struct{}
+
+var _ cciptypes.CommitPluginCodec = (*CommitPluginCodecV1)(nil)
+
+func NewCommitPluginCodecV1() *CommitPluginCodecV1 {
+	return &CommitPluginCodecV1{}
+}
+
+func (c *CommitPluginCodecV1) Encode(ctx context.Context, report cciptypes.CommitPluginReport) ([]byte, error) {
+	s := &bcs.Serializer{}
+	bcs.SerializeSequenceWithFunction(report.PriceUpdates.TokenPriceUpdates, s, func(s *bcs.Serializer, item cciptypes.TokenPrice) {
+		sourceToken := aptos.AccountAddress{}
+		err := sourceToken.ParseStringRelaxed(string(item.TokenID))
```

Comment:
nit: bit of terminology but the token prices posted ON Aptos (i.e, where this codec will be used) live on the destination chain (in this case Aptos) so maybe `destToken` makes more sense? Not sure if we use this kinda name elsewhere as well.